### ISSUE TITLE
Special sections (e.g., appendix) before chapters are not properly parsed

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#1188]
Special sections, such as appendix, that appear before chapters in the book doctype are not properly parsed (when compared to AsciiDoc Python).

Example of what doesn't work in Asciidoctor, but does work in AsciiDoc Python:

```
= Document Title
:doctype: book

[appendix]
= Appendix

appendix content

== Chapter 01

chapter 01 content
```

Asciidoctor issues a warning for the Chapter 01 section title:

```
asciidoctor: WARNING: section title out of sequence: expected level 2, got level 1
```

Moving the appendix down a level fixes the problem in Asciidoctor

```
= Document Title
:doctype: book

[appendix]
== Appendix

appendix content

== Chapter 01

chapter 01 content
```

We may want to simply tighten the AsciiDoc spec in this case to disallow the ambiguous situation in the first example.

This issue was first reported here: https://github.com/asciidoctor/asciidoctor.js/issues/67

**Comment by mojavelinux** [comment]
This is a duplicate of #1261 and was resolved in 1.5.6.

**Comment by mojavelinux** [comment]
Actually, it was an unintended duplicate. It's more that the solution for #1261 solved this issue transitively.
